base/rational.jl

```diff
@@ -11,12 +11,16 @@ struct Rational{T<:Integer} <: Real
     den::T
 
     function Rational{T}(num::Integer, den::Integer) where T<:Integer
-        num == den == zero(T) && __throw_rational_argerror(T)
+        num == den == zero(T) && __throw_rational_argerror_zero(T)
+        if T <: Signed && hastypemin(T)
+            den == typemin(T) &&  __throw_rational_argerror_typemin(T)
```

Comment:
I'm sorry, in my previous review I was just looking at the `typemin` stuff, but actually I think this error logic here is not appropriate in general. A type could have a `typemin` without the current definition being an error. The real problem occurs when `typemin(T) == -typemin(T)` (which is typically true for bit types which use 2-complement arithmetic). So I suggest instead moving the error checking in the below line, something like: `T <: Signed && signbit(den) && signbit(-den) && throw(...)`.

Reply:
Yes, makes sense. So just delete all the `hastypemin` stuff?

Reply:
I don't think, this check is very good, because it requires construction of `-den`, which is expensive in the case of `BigInt`.

Reply:
It can be constructed only when necessary.